app/controllers/users_controller.rb

```diff
@@ -6,9 +6,12 @@ class UsersController < ApplicationController
   before_action :set_user, only: %i[show update destroy update_password]
 
   def index
-    @users = current_user.admin? ? User.all : current_user
+    @users = current_user
```

Comment:
@josephelias-rubyians  Hi, this, will conflict with the pagination. You are manually adding the paginated data for it.
Ideally, index should only be listing.
so the query should be always User.where()
conditions should change based on the role and other search parameters

Reply:
@josephelias-rubyians  Or you shouldn't show the index api to non-admin current user. Its only show method for them.